### PR TITLE
[RFR] Add search feature

### PR DESCRIPTION
**Pull request by Jean-Beru**

### CODE REVIEW

File: app/Resources/translations/messages.en.yml [hunk 1]
```diff
@@ -9,6 +9,15 @@ delete: "Delete"
 
 invalid_csrf_token: "Invalid token, please refresh your page"
 
+search:
+    find: "Find..."
```
Comment (iamluc) [inline app/Resources/translations/messages.en.yml]: I prefer "search.find" as it is easier to grep

File: src/AppBundle/Controller/SearchController.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+        if (!$form->isValid()) {
+            return $this->redirectToRoute('index');
+        }
+
+        $data = $form->getData();
+        $keyword = $data['keyword'];
+
+        $adapter = $this->get('search_manager')->getPaginatorAdapter($keyword);
+        $pagination = $this->get('knp_paginator')->paginate($adapter, $request->query->getInt('page', 1));
```
Comment (iamluc) [inline src/AppBundle/Controller/SearchController.php:43]: I know that the elastica adapter of the DatathekePagerBundle is incomplete. But did you try it ?

File: src/AppBundle/Entity/RepositoryStarListener.php [hunk 3]
```diff
@@ -26,13 +35,18 @@ public function decStarsCount(RepositoryStar $repositoryStar, LifecycleEventArgs
...
             ->where('r.id = :id')
-            ->setParameter('id', $repositoryStar->getRepository()->getId())
+            ->setParameter('id', $repository->getId())
             ->getQuery();
+        $qb->execute();
+
+        $em->refresh($repository);
 
-        return $qb->execute();
+        $this->elasticaObjectPersister->replaceOne($repository);
```
Comment (iamluc) [inline src/AppBundle/Entity/RepositoryStarListener.php:50]: Shouldn't be detected by FOSElasticaBundle automatically ?

File: app/Resources/views/search/results.html.twig [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{% extends 'layout.html.twig' %}
+
+
+{% block content %}
+
+    <div class="row alert alert-info" role="alert">
```
Comment (iamluc) [inline app/Resources/views/search/results.html.twig]: IMHO, design should be similar to the user's repositories page

**Comment by iamluc** [comment]
Thanks @Jean-Beru !
Waiting for your response about KnpPaginator, but LGTM.

File: src/AppBundle/Entity/RepositoryStarListener.php (hunk 3 above)
Comment (Jean-Beru) [inline src/AppBundle/Entity/RepositoryStarListener.php:50]: Only RepositoryStar entity is updated here. But Elastica only listen to Repository entity changes.

I tried to mark entity as "scheduledUpdate" in UOW but Doctrine refuse to update a unchanged entity. So I had to update Repository manually.

File: src/AppBundle/Controller/SearchController.php (hunk 2 above)
Comment (Jean-Beru) [inline src/AppBundle/Controller/SearchController.php:43]: Yes, I tried it. ElasticaAdapter wants an `Elastica\Query\AbstractQuery` as thrid parameter but try to call `setFrom` which is an `Elastica\Query` method.

File: app/Resources/views/search/results.html.twig (hunk 4 above)
Comment (Jean-Beru) [inline app/Resources/views/search/results.html.twig]: I've just updated results page in order to be similar with user's repository page. Some improvements my be added but in an other issue.

**Comment by iamluc** [comment]
Tests failed.
As soon it is fixed, i'll merge.

Thanks a lot.

**Comment by Jean-Beru** [comment]
Fixed and squashed!

**Comment by iamluc** [comment]
GG !

**Pull request merged**